### PR TITLE
fix: Fix for on page load action in forked application

### LINKED ISSUE #9945: [Bug]: The Forked app fails to load data on page load
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

 The Forked app fails to load data on page load
[LOOM DEMO](https://www.loom.com/share/170155be60a04a0ea01dc69d75181c80)

### Steps To Reproduce

1. fork any of the application 
2. On the canvas verify the  data binded for the widget

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

> Because of the default actionId was missing in layoutOnLoad action inside page layout for the forked application, forked application on page action was not getting executed. This PR assigns the newly created action id as the defaultActionId. 

Fixes #9945 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Junit TC
> Manual test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ExamplesOrganizationClonerCEImpl.java [hunk 1]
```diff
@@ -385,7 +385,9 @@ private boolean updateOnLoadActionsWithNewActionIds(Map<String, String> actionId
         for (final Set<DslActionDTO> actionSet : layout.getLayoutOnLoadActions()) {
             for (final DslActionDTO actionDTO : actionSet) {
                 if (actionIdsMap.containsKey(actionDTO.getId())) {
-                    actionDTO.setId(actionIdsMap.get(actionDTO.getId()));
+                    final String srcActionId = actionDTO.getId();
+                    actionDTO.setId(actionIdsMap.get(srcActionId));
+                    actionDTO.setDefaultActionId(actionIdsMap.get(srcActionId));
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ExamplesOrganizationClonerCEImpl.java:390]: Earlier this statement was missing which was causing the issue as defaultActionId was not getting updated correctly

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ExamplesOrganizationClonerCEImpl.java:390]: Sorry, a little out of the loop here, what's a default action id?

Comment (abhvsn) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ExamplesOrganizationClonerCEImpl.java:390]: When we implemented the git, every resource is going to have a default resource Id from where this is originated. This is useful for 2 reasons,
1. To minimise the DB calls (In current case if the defaultActiuonId is not saved in layoutOnLoadAction then we have to make a call to newActionService to find out what is the defaultActionId for this)
2. Connect resources across branches

In git client is going to be aware of single ID for each resource.

Comment (abhvsn) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ExamplesOrganizationClonerCEImpl.java:390]: So while responding to client we just replace the actionId with defaultActionId in current case, considering client don't have to worry about different IDs we have in server side

**Review by sharat87 — APPROVED** [review]

**Pull request merged**